zokrates_core/src/semantics.rs

```diff
@@ -1794,16 +1794,16 @@ impl<'ast, T: Field> Checker<'ast, T> {
                 let assignee = self.check_assignee(assignee, module_id, types)?;
                 let e = self.check_expression(expression, module_id, types)?;
 
-                let e = FieldElementExpression::try_from_typed(e).map_err(|e| ErrorInner {
-                    span: Some(span),
-                    message: format!(
-                        "Expected right hand side of an assembly assignment to be of type field, found {}",
-                        e.get_type(),
-                    ),
-                })?;
-
                 match constrained {
                     true => {
+                        let e = FieldElementExpression::try_from_typed(e).map_err(|e| ErrorInner {
+                            span: Some(span),
+                            message: format!(
+                                "Expected right hand side of an assembly assignment to be of type field, found {}",
```

Comment:
This is confusing, did you mean something like "a field element or a complex type containing only field elements"

Reply:
This is the case of `<==` where we do expect only `field`